### PR TITLE
Introduce discovery domain

### PR DESCRIPTION
**Pull request by PetarTodorovv**
**Description**
Introduce new discovery domain.

Changes proposed in this pull request:
- Added new gateway for the discovery domain and host(s) + additional resources like virtual services, secrets, etc.
- Attached the new discovery domain to the "ord-external-certs" oathkeeper rule
- Adjusted configuration and added local setup certificates - TLS And CA certs

**Pull Request status**
- [x] Implementation
- [x] [N/A] Unit tests
- [x] [N/A] Integration tests
- [ ] `chart/compass/values.yaml` is updated 
- [x] [N/A] Mocks are regenerated, using the automated script

### CODE REVIEW

**Comment by PetarTodorovv** [comment]
reset-vm

File: chart/compass/values.yaml [hunk 1]
```diff
@@ -396,6 +396,16 @@ global:
     tokenDataHeader: "Connector-Token"
     certificateDataHeader: "Certificate-Data"
   istio:
+    discoveryMtlsGateway:
+      name: "compass-discovery-gateway"
+      namespace: "compass-system"
+      host: compass-discovery-gateway
```
[the remark this answers is not in the document]
Comment (PetarTodorovv) [reply chart/compass/values.yaml:402]: we could rename it, I just followed the current naming convention similar to the other gateways.

File: chart/compass/values.yaml [hunk 2]
```diff
@@ -427,6 +437,10 @@ global:
     serviceAccountTokenJWKS: https://kubernetes.default.svc.cluster.local/openid/v1/jwks
   ingress:
     domainName: "local.kyma.dev"
+    discoveryDomain:
+      name: "local.discovery.api"
```
[the remark this answers is not in the document]
Comment (PetarTodorovv) [reply chart/compass/values.yaml:441]: From all of the comments if I understand it correctly your suggestion is to rename the gateway to `discovery` and change the host to `api.local` which will result in URL - `https://discovery.api.local/...`(in local setup)?
In productive setup we'll have `https://discovery.dev.discovery.api.sap/...`

**Review by DimitarPetrov — DISMISSED** [review]

**Review by la4ezar — APPROVED** [review]

**Comment by PetarTodorovv** [comment]
reset-vm

**Comment by PetarTodorovv** [comment]
reset-vm

**Pull request merged**